classwork/classwork5.2/main.cpp

```diff
@@ -0,0 +1,58 @@
+#include "MegaDataPool.h"
+#include <iostream>
+#include <thread>
+#include <cassert>
+#include <mutex>
+#include <print>
+
+std::mutex io_mutex;
+
+void usePool(int threadID) {
+    try {
+        MegaData* data = acquireFromPool();
```

Comment:
it is not a good idea to enclose the whole scope with try/catch. It is better to localise scope. For example we know that lines 15-30 and 35-38 should not throw exceptions